### PR TITLE
CentrifugeJS: Integrate metadata pinning

### PR DESCRIPTION
**Pull request by sophialittlejohn**
### Description

This pull request...

- #951 

Moves metadata pinning to cent-js

### Approvals

- [ ] Dev

### CODE REVIEW

File: centrifuge-app/lambda/pinFile.js [hunk 1]
```diff
@@ -24,7 +24,6 @@ const ipfsHashToURI = (hash) => `ipfs://ipfs/${hash}`
 const handler = async (event) => {
   try {
     const { fileDataUri, fileName } = JSON.parse(event.body)
```
Comment (onnovisser) [inline centrifuge-app/lambda/pinFile.js:26]: wish we could get rid of the `fileName` somehow. it's only used to store a temp file. could we just randomly generate it?

File: centrifuge-js/src/CentrifugeBase.ts [hunk 2]
```diff
@@ -39,6 +39,7 @@ export type Config = {
   centrifugeSubqueryUrl: string
   altairSubqueryUrl: string
   metadataHost: string
+  pinFile?: (values: any) => Promise<{ uri: string; ipfsHash: string }>
```
Comment (onnovisser) [inline centrifuge-js/src/CentrifugeBase.ts:42]: could `values` just be the file and have it be type File? Or string I guess as it's now dataUri

File: centrifuge-js/src/modules/metadata.ts [hunk 3]
```diff
@@ -10,6 +10,45 @@ export function getMetadataModule(inst: Centrifuge) {
     return inst.getMetadataObservable<T>(url)
   }
 
+  function pinFile(metadata: {
+    fileDataUri?: string
+    fileName?: string
+  }): Observable<{ uri: string; ipfsHash: string }> {
```
Comment (onnovisser) [inline centrifuge-js/src/modules/metadata.ts:16]: wasn't this going to return just a string and we check if it's just an ipfs hash or a proper uri already?

Comment (sophialittlejohn) [reply centrifuge-js/src/modules/metadata.ts]: I've still left it returning an object with just the URI and in cent-js were parsing it into the full URI

**Comment by EstherCentrifuge** [comment]
if this is deployed, can we close it please? @sophialittlejohn

**Comment by sophialittlejohn** [comment]
@EstherCentrifuge that's just a PR deployment 😄 I'll close the PR and the ticket when I merge it into main (that's also when you'll be able to test it on demo and dev)

**Pull request merged**